File: alt-anns/typewiz/plain-out.ts
function nodes(t: { left: { op: string, val: number }, op: string, right: { op: string, val: number } }|{ op: string, val: number }) {
  switch t.op {
    case "node": 
      return 1 + nodes(t.left) + nodes(t.right);
    case "leaf": return 1;
    default: throw t.op;
  }
}
console.log(nodes({op: "node",
                   left: {op: "leaf", val: 1},
                   right: {op: "leaf", val: 2}}));

File: alt-anns/typpete/plain-out.py
from typing import Dict, Sequence

def vertices(t: Dict[(Sequence, object)]) -> int:
    if (t['op'] == 'node'):
        return ((vertices(t['left']) + vertices(t['right'])) + 1)
    elif (t['op'] == 'leaf'):
        return 1
    else:
        raise Exception('no')
print(vertices({'op': 'node', 'left': {'op': 'leaf', 'val': 1}, 'right': {'op': 'leaf', 'val': 2}}))
